gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAChallengeEndpoint.java

```diff
@@ -132,6 +138,15 @@ private void renderMFAPage(RoutingContext routingContext) {
             final MultiMap queryParams = RequestUtils.getCleanedQueryParams(routingContext.request());
             final String action = UriBuilderRequest.resolveProxyRequest(routingContext.request(), routingContext.request().path(), queryParams, true);
             routingContext.put(ConstantKeys.FACTOR_KEY, factor);
+
+            if (factor.is(FIDO2)) {
+                final String webAuthnLoginPath = UriBuilderRequest.resolveProxyRequest(routingContext.request(),
+                        routingContext.get(CONTEXT_PATH) + "/webauthn/login", queryParams, true);
+
+                routingContext.put("webAuthnLoginPath", webAuthnLoginPath);
+                routingContext.put("userName", endUser.getUsername());
+            }
```

Comment:
Do we still need this ? as the MFA FIDO 2 challenge will be done in the `/mfa/challenge` page ?

Reply:
I think we need this, we will discuss. I have written down the reason in the reply relates to `mfa_challenge.html`.